skip reencoding

salvah22 opened the pull request:
requirement exposed in related #384, slow re-encoding downloaded videos could be helped by a) gpu acceleration or b) skipping reeconding, this changes add the optional argument `--skip-reenconde` to allow the latter

benoit74 reviewed (changes requested):
Thank you!

Please add a `CHANGELOG.md` entry in `[Unreleased]` section, in a new `Added` subsection.

And I might be wrong, but it looks like videos will always be stored at `..../video.webm`. Extension is correct when we re-encode. But wrong if we skip re-encoding (source video might not be webm). While I agree it has zero user impact, we should fix this since it is quite misleading for the developer.

salvah22 commented:
I found some issue with windows encoding of .sh files, https://stackoverflow.com/questions/38905135/why-wont-my-docker-entrypoint-sh-execute

I will open a new merge request

salvah22 commented on scraper/src/youtube2zim/scraper.py:
```diff
@@ -744,12 +746,21 @@ def download_video(self, video_id, options):
+                    video_location,
+                    video_id
+                )
+                zim_path = f"videos/{video_id}/video{video_path.suffix}"
```
Sorry for the delay, I found some complexity here I was not prepared for, the solution I presented was not considering s3_storage and it certainly doesnt look that trivial anymore, I will need to give this a think over the weekend

benoit74 replied:
Yep, no worries, I'm not that surprised, it is always easy to introduce a hack for a given scenario, but significantly more complex when you want to make it work in a correct manner for all situations ... Let me know if you need some help

salvah22 replied:
think hopefully got something more coherent now, just pushed the changes,

but I recon a potential pitfal when s3_storage is being used, if skipping reencoding, would have to scan for the remote folder to find what actually is there, while the current implementation assumes self.file_format is what is actually stored, related to that I left a comment in src/youtube2zim/scraper.py line 727

benoit74 commented:
@salvah22 why are you closing? Is this just not working, you are giving up, you expected a new review ? I'm sorry if I missed something, but on my end I was waiting for news from you saying this was ready for a new review.

salvah22 commented:
@benoit74 hi, I thought it was stale and closed to not pollute, yeah I made the changes I thought were necessary and was expecting a review, I guess I didn't explicitly request a new review 😅

benoit74 reviewed (changes requested):
Yeah, sorry for the misunderstanding, I have to admit the PR also dropped from my radar.

We do expect PR authors to explicitly re-request reviews (preferably via the UI button, but in a comment it is fine as well) because otherwise it is hard to know if the commits you've pushed are ready or only WIP. Some authors push only once it is fully ready for new review, but most authors (including myself) push when it seems ready, then do one final round of checks, realize some things could be fine-tuned, push a new commit, etc ... Only causing confusion and frustration if we do the review too early.

salvah22 commented:
Hi @benoit74 

I have addressed all of the QA issues, added the mutually_exclusive_group and updated the variable name to videos_zim_path as requested, please let me know if there is something else needed to be done

salvah22 commented:
i know in gitlab there is an option i can tick to do so but cant find any similar in github, do i need to close this merge request and open a new one with the commits rebased?

salvah22 commented:
@benoit74 also, i can see some integration test failed in the cicd pipeline, is there something i should worry about?
